std/experimental/color/rgb.d

```diff
@@ -0,0 +1,772 @@
+module std.experimental.color.rgb;
+
+import std.experimental.color;
+
+import std.traits: isInstanceOf, isNumeric, isIntegral, isFloatingPoint, isSigned, isSomeChar, Unqual;
+import std.typetuple: TypeTuple;
+import std.typecons: tuple;
+
+@safe: pure: nothrow: @nogc:
+
+
+enum isRGB(T) = isInstanceOf!(RGB, T);
+
+enum RGBColorSpace
+{
+    sRGB,
+    sRGB_Gamma2_2,
+    AdobeRGB,
+    AppleRGB,
+    BestRGB,
+    BetaRGB,
+    BruceRGB,
+    CIE_RGB,
+    ColorMatchRGB,
+    DonRGB4,
+//    ECI_RGB_v2,
+    EktaSpacePS5,
+    NTSC_RGB,
+    PAL_SECAM_RGB,
+    ProPhotoRGB,
+    SMPTE_C_RGB,
+    WideGamutRGB
+}
+
+
+struct RGB(string components_, ComponentType_, bool linear_ = false, RGBColorSpace colorSpace_ = RGBColorSpace.sRGB) if(isValidComponentType!ComponentType_)
```

Comment:
You can't. Since it's not possible to operate on those packed types directly, they need to be unpacked to a format that can be operated on.
There is an unpopulated struct below this one that is intended to handle those packed types. It will perform packing/unpacking when assigned to/from other colour types that are operable.